packages/stylist/src/style_src.rs

```diff
@@ -50,22 +50,47 @@ pub struct StyleSource<'a> {
     _marker: PhantomData<&'a ()>,
 
     manager: Option<StyleManager>,
+    #[cfg(debug_assertions)]
+    pub(crate) location: String,
 }
 
 impl StyleSource<'_> {
-    pub(crate) fn try_to_sheet(&self) -> Result<Sheet> {
+    #[cfg(debug_assertions)]
+    #[track_caller]
+    pub(crate) fn get_location() -> String {
+        let caller_loc = std::panic::Location::caller();
+        // follow css class name rules:
+        // -?[_a-zA-Z]+[_a-zA-Z0-9-]*
+        let mut approximate_file: String = caller_loc
+            .file()
+            .chars()
+            .map(|c| if c.is_ascii_alphabetic() { c } else { '-' })
+            .collect();
+        // max len of 32 character, mostly sufficient
+        let mut split = approximate_file.len().saturating_sub(32);
+        while approximate_file.as_bytes().get(split) == Some(&b'-') {
+            split += 1;
+        }
+        // at most 32 characters. Long enough?
+        let file = approximate_file.split_off(split);
+        format!("{}-{}", file, caller_loc.line())
+    }
+
+    pub(crate) fn try_into_sheet(self) -> Result<(Sheet, Option<StyleManager>)> {
```

Comment:
This approach of "implicitly" registering manager is kind of "unclean".

Which is why I limited the manager instance to be used when it has no other ways of gaining a manager context (directly passed into elements).

I think `Style::new` should not pick up manager from `StyleSource` or it might be surprising to users of why it's registered in an unexpected manager.

Reply:
The only way to set the `StyleManager` is with the `#[doc(hidden)] pub fn with_manager` method which is used in the macros. The change to a single method is to avoid a clone. `try_to_sheet(&self) -> try_into_sheet(self)` by not passing the manager separately. This changes no visible behaviour, as before the Style would be created by `new_with_manager`.

Reply:
> https://github.com/futursolo/stylist-rs/pull/63/files#diff-07233fb7b95c2739bc301ecb64f6957b1b23dad3e7fe25f327549940d49d7799R87

I think the manager instance in the style source is picked up by `Style::new`?

My previous comment is suggesting that `Style::new()` should always use the global instance (ignore instance in the style source).

In addition, Both Manager and Sheet are Rc or Arc'ed instances, cloning them are not expensive.

Reply:
Well, there a mismatch between `Style::new()` and [the old `to_style`](https://github.com/futursolo/stylist-rs/pull/63/files#diff-dfec2c8a253834da0389dcdb23be6d2ab047e5238709dbd25bb1e364e73e7c9cL67-L68)? I would expect that `<div class={some_style_source}>` and `let style = Style::new(style_source); <div class={style}>` produce equivalent results.

The `css!` macro generated by `styled_component` puts a manager into the `StyleSource`, but the only way for a user to observe this is by going straight to `Classes`, there is no other way to retrieve it. I think that's even more surprising.

What I meant was that the only way to put a manager into a `StyleSource` is to use the generated `css!` macro and I don't think it's surprising that it remembers the context.

In any case, I was just adding it to the PR because I touched part of the code anyway, I can separate it.

Reply:
It's an API that is outside of `stylist::yew`, so it should not take Yew contexts into consideration.

In other words, `Style::new(css!())` should be reliably registering to the global instance, same as `Style::new("with a string")` and `style!("string or stylesheet.")`.

If one wants to acknowledge style manager from yew context, they should either use `styled_component` in it's intended way (pass directly to class), or use `use_style!` / `use_style`.

Reply:
The only way to add a `StyleManager` to a `StyleSource` is by using the yew-specific `css!` macro that is generated by `styled_component`, so this doesn't break the API barrier. Will still remove this change from the PR, so it can move ahead.